### DISCUSSION TITLE
Setting color scheme for a pane from Lua

### DISCUSSION THREAD

**Discussion by kontza** [wezterm/wezterm#4744] [General]
Hi,

Is it possible to set pane's background color/color scheme from Lua?

**Comment by nekowinston** [comment]
Configuration can be overridden per-window, not per-pane; you can change the active color scheme based on which pane is focused, though. See https://github.com/wez/wezterm/discussions/4260

**Comment by kontza** [comment]
Ok, thanks. I was just looking for an alternative way of changing pane background for SSH to a remote host. I know that I could do it so, that the remote end's bashrc etc would print out an ANSI escape codes to change colors, but I like my current setup where everything happens client side.

My current setup for changing pane background is as follows:

1. SSH config 'runs' a LocalCommand, a script that runs https://github.com/drinchev/phook to capture SSH's PID.
2. Then the script prints out an ANSI escape sequence based on the host where I'm connecting to. This sets the active pane's background color.
3. When SSH exists, i.e. connection closes, phook runs another script that restores the pane background to the default one.

The above works, but as it consists of multiple moving parts, I was looking for a way to have all this happen within Wezterm via Lua and events.

**Reply by wez** [reply]
Take a look a User Vars. https://wezfurlong.org/wezterm/recipes/passing-data.html#user-vars has some examples.
You could set a user var to specify some metadata per-pane, then apply that in the `update-status` event.

**Reply by kontza** [reply]
Ok, thanks. One, final, question: can Lua change the pane background color? Wezterm itself can do it, as I'm currently using `OSC 1337 SetColors` to set it. But is there an API in Lua-side for it?

**Reply by wez** [reply]
Configuration is per-window, rather than per-pane, so you can't set it via configuration from lua.
However, you can use https://wezfurlong.org/wezterm/config/lua/pane/inject_output.html to inject escape sequences into *local* (not multiplexer!) panes.

**Comment by kontza** [comment]
Ok, thanks.

Juha R.


pe 5.1.2024 klo 15.38 Wez Furlong ***@***.***> kirjoitti:

**Comment by kontza** [comment]
For anyone interested, this is what I finally came up with:
https://gist.github.com/kontza/7ee30f5ed7ae949e90bd03bc0d8bc63d
